extended/src/main/java/apoc/ExtendedApocConfig.java

```diff
@@ -0,0 +1,172 @@
+    public static final String APOC_UUID_FORMAT = "apoc.uuid.format";
+    public enum UuidFormatType { hex, base64 }
+
+    private final Log log;
+
+    private Configuration config;
+
+    private static ExtendedApocConfig theInstance;
+
+    private ExtendedApocConfig.LoggingType loggingType;
+    private SimpleRateLimiter rateLimiter;
+
+    /**
+     * keep track if this instance is already initialized so dependent class can wait if needed
+     */
+    private boolean initialized = false;
+
+    private static final String DEFAULT_PATH = ".";
+    private static final String CONFIG_DIR = "config-dir=";
+
+    public ExtendedApocConfig(LogService log, GlobalProcedures globalProceduresRegistry) {
+        this.log = log.getInternalLog(ApocConfig.class);
+        theInstance = this;
+
+        // expose this config instance via `@Context ApocConfig config`
+        globalProceduresRegistry.registerComponent((Class<ExtendedApocConfig>) getClass(), ctx -> this, true);
+        this.log.info("successfully registered ExtendedApocConfig for @Context");
+    }
+
+    @Override
+    public void init() {
+        log.debug("called init");
+        // grab NEO4J_CONF from environment. If not set, calculate it from sun.java.command system property
+        String neo4jConfFolder = System.getenv().getOrDefault("NEO4J_CONF", determineNeo4jConfFolder());
+        System.setProperty("NEO4J_CONF", neo4jConfFolder);
+        log.info("system property NEO4J_CONF set to %s", neo4jConfFolder);
+        loadConfiguration();
+        initialized = true;
+    }
+
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    protected String determineNeo4jConfFolder() {
+        String command = System.getProperty(SUN_JAVA_COMMAND);
+        if (command == null) {
+            log.warn("system property %s is not set, assuming '.' as conf dir. This might cause `apoc.conf` not getting loaded.", SUN_JAVA_COMMAND);
+            return DEFAULT_PATH;
+        } else {
+            final String neo4jConfFolder = Stream.of(command.split("--"))
+                    .map(String::trim)
+                    .filter(s -> s.startsWith(CONFIG_DIR))
+                    .map(s -> s.substring(CONFIG_DIR.length()))
+                    .findFirst()
+                    .orElse(DEFAULT_PATH);
+            if (DEFAULT_PATH.equals(neo4jConfFolder)) {
+                log.info("cannot determine conf folder from sys property %s, assuming '.' ", command);
+            } else {
+                log.info("from system properties: NEO4J_CONF=%s", neo4jConfFolder);
+            }
+            return neo4jConfFolder;
+        }
+    }
+
+    /**
+     * use apache commons to load configuration
+     * classpath:/apoc-config.xml contains a description where to load configuration from
+     */
+
+    protected void loadConfiguration() {
+        try {
+
+            URL resource = getClass().getClassLoader().getResource("apoc-config.xml");
```

Comment:
As extended is now separate from core, is there any concern with them using the same config file?

Reply:
Good question... As extended still use config from the core file, I guess it should be alright. It could be cleaner to have the extended config in a separate file, but on the other hand it will be one more thing to do for people migrating from extended 4.x to extended 5.x. What do you think? Should we ask the others in the team for feedback regarding this?

Reply:
Hmm I think that makes sense remembering it uses the common package actually. I am happy for it to continue to use the same file, but feel free to ask the team for feedback, there might be something else we haven't thought about.